requests_toolbelt/adapters/appengine.py

```diff
@@ -44,6 +49,57 @@ def __init__(self, validate_certificate=True, *args, **kwargs):
         super(AppEngineAdapter, self).__init__(self, *args, **kwargs)
 
     def init_poolmanager(self, connections, maxsize, block=False):
-        self.poolmanager = gaecontrib.AppEngineManager(
-            validate_certificate=self._validate_certificate
-        )
+        self.poolmanager = _AEPoolManager(self._validate_certificate)
+
+
+class _AEPoolManager(object):
+    """Implements urllib3's PoolManager API."""
+
+    def __init__(self, validate_certificate=True):
+        self.validate_certificate = validate_certificate
+
+    def connection_from_url(self, url):
+        return _AEConnection(self, url)
+
+    def clear(self):
+        pass
+
+
+class _AEConnection(object):
+    """Implements urllib3's HTTPConnectionPool API's urlopen().
+
+    This Connection's urlopen() is called with a host-relative path,
+    so in order to properly support opening the URL, we need to store the full URL
+    when we are constructed from the Manager.
+    """
+
+    def __init__(self, pool_manager, url):
+        self.pool_manager = pool_manager
+        self.url = url
+
+    def urlopen(self, method, url, body=None, headers=None, retries=None,
+                redirect=True, assert_same_host=True, timeout=timeout.Timeout.DEFAULT_TIMEOUT,
+                pool_timeout=None, release_conn=None, **response_kw):
+        # Jump through the hoops necessary to call AppEngineManager's API.
+        appengine_manager = gaecontrib.AppEngineManager(
```

Comment:
Good idea, fixed. Though really, the way this code gets used, I believe urlopen was only called once per connection anyway. (Though see note in next comment, as that assumption could change.)